Modules/Core/include/selxComponentBase.h

```diff
@@ -54,6 +54,9 @@ class ComponentBase
 
   typedef std::map< std::string, std::string > InterfaceCriteriaType;
 
+  // Will be implemented in SuperElastixComponent so that any interface can add a GetComponentName method without the need to implement that in each component individually.
+  //virtual const typename std::string GetComponentName() = 0;
```

Comment:
I don't think "typename" is necessary here. The function should probably be declared with "const" at the end:  virtual std::string GetComponentName() const = 0;

Reply:
this commented line was an unsuccessful attempt to fix the warning about missing overrides. I simply copied the declaration from SuperElastixComponent. So your comment (which seems valid to me) would apply to that file, and other files implementing this method.